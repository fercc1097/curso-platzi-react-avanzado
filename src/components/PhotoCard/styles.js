/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';
import { fadeIn } from '../../styles/animation';

export const ImgWrapper = styled.div`
  border-radius: 10px;
  height: 0;
  overflow: hidden;
  padding: 56.25% 0 0 0;
  display: block;
  position: relative;
  width: 100%;
`;

export const Img = styled.img`
  box-shadow: 0 10px 14px rgba(0, 0, 0, 0.2);
  height: 100%;
  width: 100%;
  object-fit: cover;
  position: absolute;
  top: 0;
  ${fadeIn()}
`;

export const Article = styled.article`
  min-height: 200px;
`;
