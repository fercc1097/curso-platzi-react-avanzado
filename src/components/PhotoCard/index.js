/* eslint-disable no-unused-expressions */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/prefer-default-export */
import React from 'react';
import { Link } from '@reach/router';
import { ImgWrapper, Img, Article } from './styles';
import { FavButton } from '../FavButton';
import { useLocalStorage } from '../../hooks/useLocalStorage';
import { useNearScreen } from '../../hooks/useNearScreen';
import { ToggleLikeMutation } from '../../container/ToggleLikeMutation';

const DEFAULT_IMAGE =
  'https://images.unsplash.com/photo-1518001589401-1743b61d1def?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60';

export const PhotoCard = ({ id, likes = 0, src = DEFAULT_IMAGE }) => {
  const [show, element] = useNearScreen();
  const key = `like-${id}`;
  const [liked, setLiked] = useLocalStorage(key, false);

  return (
    <Article ref={element}>
      {show && (
        <>
          <Link to={`/detail/${id}`}>
            <ImgWrapper>
              <Img src={src} alt='an animal' />
            </ImgWrapper>
          </Link>
          <ToggleLikeMutation>
            {(toggleLike) => {
              const handleFavClick = () => {
                !liked &&
                  toggleLike({
                    variables: {
                      input: { id },
                    },
                  });
                setLiked(!liked);
              };
              return (
                <FavButton
                  liked={liked}
                  likes={likes}
                  onClick={handleFavClick}
                />
              );
            }}
          </ToggleLikeMutation>
        </>
      )}
    </Article>
  );
};
