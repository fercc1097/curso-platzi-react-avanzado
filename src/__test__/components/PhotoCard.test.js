import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import { PhotoCard } from '../../components/PhotoCard/index';

describe('<PhotoCard />', () => {
  test('Render del componente Footer ', () => {
    const photoCard = mount(<PhotoCard />);
    expect(photoCard.length).toEqual(1);
  });
});

describe('FavButton Snapshot', () => {
  test('Comprobar la Ui del componente Footer', () => {
    const photoCard = create(<PhotoCard />).toJSON();
    expect(photoCard).toMatchSnapshot();
  });
});
