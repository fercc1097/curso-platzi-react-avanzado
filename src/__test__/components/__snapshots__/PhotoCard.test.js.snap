// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavButton Snapshot Comprobar la Ui del componente Footer 1`] = `
<article
  className="sc-dlnjPT bipitr"
/>
`;
